libgobuster/libgobuster.go

```diff
@@ -33,6 +34,7 @@ type Gobuster struct {
 	context          context.Context
 	requestsExpected int
 	requestsIssued   int
+	cachedResults    []Result
```

Comment:
This is a bad idea. This will blow out memory big time for huge result sets. Data should be streamed out as it's captured rather that cached and then serialised in one hit.